### PR TITLE
bpo-19468: delete unnecessary instance check in importlib.reload()

### PR DESCRIPTION
**Pull request by furkanonder**
https://bugs.python.org/issue19468

Automerge-Triggered-By: @brettcannon

### CODE REVIEW

**Review by eamanu — APPROVED** [review]
LGTM

**Comment by eamanu** [comment]
IMO news could be skipped.

**Review by ericsnowcurrently — APPROVED** [review]
LGTM

I expect there could possibly be some deep corner cases (via `_bootstrap._find_spec()` and `_bootstrap._exec()` that would change behavior here, but find it highly unlikely that, if such cases even exist, anyone relies on them. :smile:

**Comment by furkanonder** [comment]
@brettcannon

**Comment by brettcannon** [comment]
LGTM, although I think a news entry is still appropriate.

**Review by brettcannon — CHANGES_REQUESTED** [review]
Needs a news entry as it changes the semantics of the function.

**Comment by brettcannon** [comment]
@furkanonder please read the instructions from the bot on how to signal you're ready for another review.

**Comment by furkanonder** [comment]
> @furkanonder please read the instructions from the bot on how to signal you're ready for another review.

Sorry, I missed it. I have made the requested changes; please review again :)

File: Misc/NEWS.d/next/Core and Builtins/2020-05-30-23-18-35.bpo-19468.S-TA7p.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Delete unnecessary instance check in importlib.reload().
```
Comment (brettcannon) [inline Misc/NEWS.d/next/Core and Builtins/2020-05-30-23-18-35.bpo-19468.S-TA7p.rst]:
```suggestion
Delete unnecessary instance check in importlib.reload().
Patch by Furkan Önder.
```

**Review by brettcannon — APPROVED** [review]

**Pull request merged**